packages/server/src/adapters/content-handlers/selectContentHandlerOrUnsupportedMediaType.ts

```diff
@@ -0,0 +1,20 @@
+import { TRPCError } from '../../@trpc/server';
+import type { BaseContentTypeHandler } from '../../@trpc/server/http';
+
+export function selectContentHandlerOrUnsupportedMediaType<
+  THandlerOpts,
+  THandler extends BaseContentTypeHandler<THandlerOpts>,
+>(handlers: THandler[], opts: THandlerOpts) {
+  const handler = handlers.find((handler) => handler.isMatch(opts));
+  if (!handler) {
+    return [
+      undefined,
```

Comment:
null is nicer imo but that's a nit, feel free to ignore me

Reply:
I don't mind changing this, downstream the `error?: TRPCError` property is optional so we support undefined in quite a clean way, but happy to tweak if you'd like

Reply:
Going to go with undefined just because of the TS benefits elsewhere :)